src/main/java/com/redhat/foreman/ForemanComputer.java

```diff
@@ -63,7 +65,12 @@ static void terminateForemanComputer(Computer c) throws IOException, Interrupted
             Node node = fc.getNode();
             if (node != null) {
                 ForemanSharedNode sharedNode = (ForemanSharedNode)node;
+                if (sharedNode.getChannel() != null) {
+                    sharedNode.getChannel().close();
+                }
                 sharedNode.terminate();
+                Jenkins.getInstance().removeNode(sharedNode);
```

Comment:
`AbstractCloudSlave.terminate()` calls `Jenkins.getInstance().removeNode(this);` already.